src/DGmethods/test/Sphere/advection_sphere.jl

```diff
@@ -0,0 +1,290 @@
+    # Set up the information callback
+    timer = [time_ns()]
+    cbinfo = GenericCallbacks.EveryXWallTimeSeconds(10, mpicomm) do (s=false)
+        if s
+            timer[1] = time_ns()
+        else
+            run_time = (time_ns() - timer[1]) * 1e-9
+            (min, sec) = fldmod(run_time, 60)
+            (hrs, min) = fldmod(min, 60)
+            @printf(io, "----\n")
+            @printf(io, "simtime =  %.16e\n", ODESolvers.gettime(lsrk))
+            @printf(io, "runtime =  %03d:%02d:%05.2f (hour:min:sec)\n", hrs, min, sec)
+            @printf(io, " Δmass =  %.16e\n", abs(weightedsum(Q) - weightedsum(Qe)) / weightedsum(Qe) )
+        end
+        nothing
+    end
+
+    # Set up the information callback
+    cbmass = GenericCallbacks.EveryXSimulationSteps(1000) do
+        @printf(io, "----\n")
+        @printf(io, " Δmass =  %.16e\n", abs(weightedsum(Q) - weightedsum(Qe)) / weightedsum(Qe) )
+        nothing
+    end
+
+    step = [0]
+    mkpath("vtk")
+    cbvtk = GenericCallbacks.EveryXSimulationSteps(1000) do (init=false)
+        outprefix = @sprintf("vtk/advection_sphere_%dD_mpirank%04d_step%04d",
+                             3, MPI.Comm_rank(mpicomm), step[1])
+        @printf(io, "----\n")
+        @printf(io, "doing VTK output =  %s\n", outprefix)
+        DGBalanceLawDiscretizations.writevtk(outprefix, Q, spacedisc, ("ρ", ))
+        step[1] += 1
+        nothing
+    end
+    #------------Set Callback Info--------------------------------#
+
+    #Perform Time-Integration
+    solve!(Q, lsrk; timeend=timeend, callbacks=(cbinfo, cbmass, cbvtk))
+
+    # Print some end of the simulation information
+    if integration_testing
+        error = euclidean_distance(Q, Qe) / norm(Qe)
+        Δmass = abs(weightedsum(Q) - weightedsum(Qe)) / weightedsum(Qe)
+        @info @sprintf """Finished
+        error = %.16e
+        Δmass = %.16e
+        """ error Δmass
+    end
+
+    #return diagnostics
+    return (error, Δmass)
+end
+#}}} Main
+
+#{{{ Run Script
+function run(mpicomm, Nhorizontal, Nvertical, N, timeend, DFloat, dt, ArrayType)
+    Rrange=range(DFloat(1); length=Nvertical+1, stop=2)
+    topl = StackedCubedSphereTopology(mpicomm,Nhorizontal,Rrange; boundary=(0,0))
+    (error, Δmass) = main(mpicomm, DFloat, topl, N, timeend, ArrayType, dt)
+end
+#}}} Run Script
+
+#{{{ Run Program
+using Test
+let
+    MPI.Initialized() || MPI.Init()
+    Sys.iswindows() || (isinteractive() && MPI.finalize_atexit())
+    mpicomm=MPI.COMM_WORLD
+
+    if MPI.Comm_rank(mpicomm) == 0
+        ll = uppercase(get(ENV, "JULIA_LOG_LEVEL", "INFO"))
+        loglevel = ll == "DEBUG" ? Logging.Debug :
+        ll == "WARN"  ? Logging.Warn  :
+        ll == "ERROR" ? Logging.Error : Logging.Info
+        global_logger(ConsoleLogger(stderr, loglevel))
+    else
+        global_logger(NullLogger())
+    end
+
+    #Perform Integration Testing for three different grid resolutions
+    if integration_testing
```

Comment:
You should define a non integration testing version which will run faster for CI. I'd recommend using random numbers to "hash" the numerics, but as we discussed on Friday you're welcome to use whatever approach you'd like.

Goal here is that we want a fast way to check to see if something major has changed.